### ISSUE TITLE
UI Feedback

### ISSUE THREAD

**Issue by RubenFro** [hicetnunc2000/hicetnunc#249]
I think it would be helpful to add some feedback on the buttons.
I'm not really sure how the connection with the wallet works, and if there are some listeners to make it possible, but for example it would be nice to:

- When buying, clicking on the purchase button should make the link change to an inactive state (making it gray, lower opacity, or also adding some sort of animated "...") while it's waiting for the wallet approval. Had quite a few times where nothing happened and had to reload the page, sometimes it's instantaneous, while often it's a couple of seconds. Having a feedback can let you know that we've sent the request to the wallet and we're waiting for it.

- When minting a piece, there's very little feedback when the procedure has been completed. Also, after a successful mint we don't need to have the submit button stay on screen. I would leave all the fields as they are but replacing the submit button with "Done!" or something like it. An extra addition could be a "Would you like to curate this piece now?" linking to the objkt page curation.

As I said, I'm not sure how the wallet API works, so not sure what those changes would require.

**Comment by crzypatchwork** [comment]
yes the idea is to have a loading circle for every operation that takes long, awaiting the operation confirmation

**Comment by tarwin** [comment]
![image](https://user-images.githubusercontent.com/646149/112743190-fa521500-8f49-11eb-8c2e-b620351426b5.png)

I don't think permissions for camera or microphone work on a per-request basis. Once you have granted access you always have access unless someone manually turns it off.

Geolocation works in a different way I think, where you can get it only when specifically requested / approved using the Permissions API (https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API). From the docs it looks like browsers were thinking of rolling all permissions into this API but currently (since like Chrome 44?) it has only been Geolocation and Notifications.

**Comment by tarwin** [comment]
https://user-images.githubusercontent.com/646149/112743389-ba8c2d00-8f4b-11eb-9efa-d31dee51bf13.mp4

Even something subtle like this for "vote" (? is it even called this - there isn't even alt text) would be good. In the code could be enabled/disabled around the `await` for getting wallet info. (Will try myself if I can get the codebase running)

**Comment by tarwin** [comment]
Added code to do what I showed above. Hopefully it is up to scratch.

https://github.com/hicetnunc2000/hicetnunc/pull/278

I guess we should really add some kind of waiting state to buttons such as "Collect" and "Mint" as well?

**Comment by tarwin** [comment]
Sorry my previous pull requests was a total failure at React - and probably at adding a pull request. I have done it the correct (hook) way now, so hopefully should all be good.

Also reused the <Loader /> component to keep things simple and on brand.

https://github.com/hicetnunc2000/hicetnunc/pull/289
